inc/manifest.php

```diff
@@ -30,130 +42,84 @@ function load_asset_file( $path ) {
-			return;
-		}
+	// Next, try hashing the contents of the asset manifest file (if available).
+	// We don't hash the asset file itself because we may not know where to find
+	// those on disk -- this does mean a new asset file invalidates ALL deployed
+	// assets, but that is still preferable to versioning based on the Altis
+	// revision constant (our last resort).
+	static $manifest_hashes = [];
 
-		// Build a list of problem hosts.
-		var hosts = scriptsWithErrors.reduce(
-			function( hosts, script ) {
-				var src = script.getAttribute( 'src' );
-				if ( ! src || ! /https:\/\/localhost/i.test( src ) ) {
-					return hosts;
-				}
-				src = src.replace( /^(https:\/\/localhost:\d+).*$/i, '$1' );
-				hosts[ src ] = true;
-				return hosts;
-			},
-			{}
-		);
-		hosts = Object.keys( hosts );
-
-		// Build the error markup.
-		const messageHTML = [
-			'<strong>Error loading scripts from localhost!</strong>',
-			'<br>',
-			'Ensure that ',
-			( hosts.length > 1 ? 'these hosts are ' : 'this host is ' ),
-			'accessible, and that you have accepted any development server SSL certificates:',
-			'<ul>',
-			hosts.map( host => '<li><a target="_blank" href="' + host + '">' + host + '</a></li>' ).join( '' ),
-			'</ul>'
-		].join( '' );
-
-		notices.createErrorNotice( messageHTML, { __unstableHTML: true } );
+	if ( isset( $manifest_hashes[ $manifest_path ] ) ) {
+		return $manifest_hashes[ $manifest_path ];
 	}
 
-	// Set up processErrors to run 1 second after page load.
-	document.addEventListener( 'DOMContentLoaded', function() {
-		setTimeout( processErrors, 1000 );
-	} );
-} )();
-</script>
-	<?php
-}
+	if ( file_exists( $manifest_path ) ) {
+		$manifest_hash = md5_file( $manifest_path );
+		if ( $manifest_hash ) {
+			$manifest_hashes[ $manifest_path ] = $manifest_hash;
+			return $manifest_hashes[ $manifest_path ];
+		}
+	} else {
+		// Set "null" to prevent subsequent file_exists checks during this request.
+		$manifest_hashes[ $manifest_path ] = null;
+	}
 
-/**
- * Inject an onerror attribute into the rendered script tag for any script
- * loaded from localhost with an HTTPS protocol.
- *
- * @param string $tag    The HTML of the script tag to render.
- * @param string $handle The registered script handle for this tag.
- * @param string $src    The src URI of the JavaScript file this script loads.
- * @return string The script tag HTML, conditionally transformed.
- */
-function add_onerror_to_scripts( string $tag, string $handle, string $src ) : string {
-	if ( ! preg_match( '#https://localhost#', $src ) ) {
-		return $tag;
+	// Finally, use the Altis deployment revision when available. This is a
+	// heavy-handed solution which will expire uncached assets whenever a new
+	// deploy goes out, so we recommend using a hashed asset filename instead.
+	if ( function_exists( 'Altis\\get_environment_codebase_revision' ) ) {
+		return Altis\get_environment_codebase_revision();
 	}
```

Comment:
Good question. We're talking about making this loadable as an altis dependency #15 so I'm thinking this codebase is likely to move _towards_ Altis, if anything — in that context, I think the optional check here is desirable. We could introduce a filter later on, but this feels like it will solve more situations out of the box, and "work in a logical way" is the major goal of 0.4, so I'm going to land this as-is for now.